controllers/providers/pxc/applier.go

```diff
@@ -155,6 +155,18 @@ func (p *applier) DataSource() error {
 	if p.DB.Status.Status != everestv1alpha1.AppStateReady {
 		return nil
 	}
+	dbRestore := &everestv1alpha1.DatabaseClusterRestore{}
+	err := p.C.Get(p.ctx, types.NamespacedName{
+		Namespace: p.DB.Namespace,
+		Name:      p.DB.Name,
+	}, dbRestore)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if dbRestore.IsComplete(p.DB.Spec.Engine.Type) {
+		p.DB.Spec.DataSource = nil
+		return nil
+	}
```

Comment:
Does it make sense to move this logic into the `common.ReconcileDBRestoreFromDataSource` method? That way we will have less code duplication

Reply:
A good idea, thanks, it turned out that for pxc and psmdb the process is identical, so it can be simplified indeed. Updated.